constraints_file.py

```diff
@@ -59,3 +61,30 @@ def add_limit_employees_working_days_constraint(shifts: list[Shift], employees:
             shifts_employee_is_working.append(shift_combinations[key])
 
         constraint_model.Add(sum(shifts_employee_is_working) <= max_working_days)
+
+
+def add_minimum_time_between_closing_shift_and_next_shift_constraint(shifts: list[Shift], employees: list[Employee], constraint_model: cp_model.CpModel, shift_combinations: dict[ShiftCombinationsKey, IntVar], min_hours_between_shifts: datetime.timedelta) -> None:
+    for employee in employees:
+        shifts_window = 3
+        for shift1, shift2, shift3 in more_itertools.windowed(shifts, shifts_window):
```

Comment:
i can see why this function works. i must say i dont understand the flow at all. i would rather have 3 functions for the options of `None`

Reply:
ie in side the window call 3 different functions according to the window